starstream_compiler/src/ast.rs

```diff
@@ -168,10 +168,13 @@ pub enum Expr {
     NotEquals(Box<Self>, Box<Self>),
     /// `a < b`
     LessThan(Box<Self>, Box<Self>),
-    // TODO: GreaterThan, LessEq, GreaterEq
+    /// `a > b`
+    GreaterThan(Box<Self>, Box<Self>),
+    /// `a <= b`
+    LessEq(Box<Self>, Box<Self>),
+    /// `a >= b`
+    GreaterEq(Box<Self>, Box<Self>),
     // Arithmetic operators
-    /// `-a`
-    Neg(Box<Self>),
```

Comment:
Any particular reason to drop Neg? Wasm has a floating-point `neg` instruction, and integer `neg` is easy to implement as `~x + 1`

Reply:
Mostly because it was previously `~`, and when I changed that I removed it. But it probably makes sense to keep it I guess.

So I added it back. I'm not really sure how much people actually use the unary negation in general though.